internal/servers/plugin/v3/plugin.go

```diff
@@ -463,18 +465,26 @@ func (s *Server) Transform(stream pb.Plugin_TransformServer) error {
 			}
 			if err != nil {
 				close(recvRecords)
-				return fmt.Errorf("Error receiving request: %v", err)
-			}
-			if doneReading {
-				return nil
+				return status.Errorf(codes.Internal, "Error receiving request: %v", err)
 			}
 			record, err := pb.NewRecordFromBytes(req.Record)
 			if err != nil {
 				close(recvRecords)
 				return status.Errorf(codes.InvalidArgument, "failed to create record: %v", err)
 			}
 
-			recvRecords <- record
+			select {
+			case recvRecords <- record:
+			case err := <-pluginStops:
+				close(recvRecords)
+				return err
+			case <-gctx.Done():
+				close(recvRecords)
+				return gctx.Err()
```

Comment:
What error is returned here when:
1. The context is done because it was cancelled
2. The context is done because an error group Go routine returned an error
3. The context is done because all Go routines finished running

Reply:
I think this is ok as long as the error returned if the one from the transformer if exists